pipeline/matching/evaluation/analytics.py

```diff
@@ -0,0 +1,24 @@
+from typing import Dict, Iterator, List
+
+from pipeline.types import Match, MatchingInput, MatchingInternalAnalytics, User
+
+
+class CollectInternalAnalytics:
+    def __init__(self):
+        pass
+
+    def n_proposed_matches_per_user(
+        self, users: List[User], ranked_matches: Iterator[Match]
+    ) -> MatchingInternalAnalytics:
+
+        analytics: Dict = {}
```

Comment:
Have you heard of `defaultdict` before? It is part of the built-in `collections` module with Python. Might make this code cleaner because it allows for a default value when the key is not yet in the dictionary.

Up to you if you want to look it up and use that!

If you use a `defaultdict`, cast it to a regular dict when you return it, that way other keys don't get added when callers try to check invalid users.